### PR TITLE
Fixes overlap of posts when height changes

### PR DESCRIPTION
**Pull request by sudheerDev**
* Remove mutation observer and add a way to emit scroll events when height changes

### CODE REVIEW

File: src/createListComponent.js [hunk 1]
```diff
@@ -171,14 +171,20 @@ export default function createListComponent({
...
+      this._scrollCorrectionInProgress = true;
       this.setState(
         prevState => ({
           scrollDirection:
             prevState.scrollOffset >= scrollOffset ? 'backward' : 'forward',
           scrollOffset: scrollOffset,
           scrollUpdateWasRequested: true,
         }),
-        this._resetIsScrollingDebounced
+        () => {
```
Comment (sudheerDev) [inline src/createListComponent.js:183]: @enahum This is the change i was referring to for fixing position when there are lot of unread.

File: src/ItemMeasurer.js [hunk 2]
```diff
@@ -46,44 +45,163 @@ type ItemMeasurerProps = {|
...
+    if (
+      (prevProps.size === 0 && this.props.size !== 0) ||
+      prevProps.size !== this.props.size
+    ) {
+      this.positionScrollBars();
+    }
   }
 
+  positionScrollBars = (height = this.props.size, width = this.props.width) => {
+    //we are position these hiiden div scroll bars to the end so they can emit
```
Comment (sudheerDev) [inline src/ItemMeasurer.js:140]: Links here summarise on how we are triggering recalc based on scroll event when height changes

File: src/ItemMeasurer.js [hunk 3]
```diff
@@ -46,44 +45,163 @@ type ItemMeasurerProps = {|
...
+
+      this._resizeSensorShrink.current.scrollBy(
+        2 * height + 17,
+        2 * width + 17
+      );
+    } else {
+      this._resizeSensorExpand.current.scrollLeft = width + 27;
+      this._resizeSensorExpand.current.scrollTop = height + 27;
+      this._resizeSensorShrink.current.scrollTop = 2 * height + 17;
+      this._resizeSensorShrink.current.scrollLeft = 2 * width + 17;
```
Comment (saturninoabril) [inline src/ItemMeasurer.js]: nit: Can we make 27 and 17 as constant and add comment so that we know how/when to change it in the future?

**Review by saturninoabril — APPROVED** [review]

File: src/ItemMeasurer.js (hunk 3 above)
Comment (sudheerDev) [reply src/ItemMeasurer.js]: Nice catch. i thought to comment that but i forgot about it. I will add the comment for now. I am planning on making it dynamic and set the values on init later

Comment (sudheerDev) [reply src/ItemMeasurer.js]: Done.

File: src/createListComponent.js (hunk 1 above)
Comment (enahum) [reply src/createListComponent.js:183]: Ok, still not following what has to be done. Also saw the comment from @saturnino about the permalink view and wonder how will that affect

**Review by enahum — APPROVED** [review]

File: src/createListComponent.js (hunk 1 above)
Comment (sudheerDev) [reply src/createListComponent.js:183]: I removed the scrollOffset correction on componentDidUpdate as the flag `scrollUpdateWasRequested ` will be true and it triggers that offset correction causing issues. 

I am guessing this should correct it but nevertheless this makes the code predictable as this is the flow for which we need that correction in componentDidUpdate.

Whenever we update position with scrollTo we set the state value of the offset here and correct the scroll position in componentDidUpdate using the flag `scrollUpdateWasRequested ` but as this flag is not reset unless user scrolls we might actually exec that scroll correction code as side effect for other scenarios.

**Pull request merged**